canine/localization/base.py

```diff
@@ -540,15 +557,19 @@ def job_setup_teardown(self, jobId: str, patterns: typing.Dict[str, str]) -> typ
         Must call after pre-scanning inputs
         """
 
-		# generate job variable, exports, and localization_tasks arrays
-		# - job variables and exports are set when setup.sh is _sourced_
-		# - localization tasks are run when localization.sh is _run_
+        # generate job variable, exports, and localization_tasks arrays
+        # - job variables and exports are set when setup.sh is _sourced_
+        # - localization tasks are run when localization.sh is _run_
         job_vars = []
-        exports = []
-        docker_args = ['-v $CANINE_ROOT:$CANINE_ROOT']
+        exports = [
+            'export CANINE_NODE_NAME=$(curl -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/instance/name 2> /dev/null)',
+            'export CANINE_NODE_ZONE=$(basename $(curl -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/instance/zone 2> /dev/null))'
+        ]
+        docker_args = []
```

Comment:
```suggestion
        docker_args = ['-v $CANINE_ROOT:$CANINE_ROOT']
```

Don't we still need to keep this value in `docker_args`? Otherwise normal filepaths in the staging directory won't be visible in the docker

Reply:
`ro_disk` is not intended to replace the local disk download. Currently, `ro_disk` requires that the read only disk was created by a special upstream localization task (which is actually implemented in wolF), so it's impossible to simultaneously create the disk and process files on it in one task.

The `local_disk` override is still required for workflows that need to localize a large file once and be done with it (e.g. FASTQ files for aligners). 

From a UX standpoint, this is very subtle. I suspect most users will transparently use `ro_disk` via the special wolF localization task, which in turn will automatically invoke Canine mounting it in all downstream tasks via the special `gsdisk://` URL output by the localization task. Users who want the `local_disk` mode will have to explicitly enable it in overrides.

One non-urgent TODO is to have Canine be able to _delocalize_ to a separate volume (à la the special wolF localization task), which subsequently gets detached and can be mounted read-only on downstream tasks. This would be ideal for alignment pipelines, which both read and write very large files.

Reply:
>Don't we still need to keep this value in docker_args? Otherwise normal filepaths in the staging directory won't be visible in the docker

Since the Docker backend assumes a shared NFS, the fact that the backend always mounts `/mnt/nfs` should take care of this. It won't break anything to add it back here, but I removed it to cut down on redundancy.

Reply:
Yeah, but isn't this a change to the base localizer? Won't that change touch more than just docker-backend jobs?

Reply:
You are correct. I'm dumb.